terraform/api_gateway.tf.json.template.py

```diff
@@ -327,13 +327,21 @@ def for_domain(cls, domain):
                           'maximum_retry_attempts': 0
                       }
                       for function_name in (
-                          'forward_alb_logs',
-                          'forward_s3_logs',
+                          *(
+                              ('forward_alb_logs', 'forward_s3_logs')
+                              if config.enable_log_forwarding else
+                              ()
+                          ),
+                          *(
+                              ('notify',)
+                              if config.enable_monitoring else
+                              ()
+                          )
                       )
                   }
               }
           ]
-          if config.enable_log_forwarding else
+          if config.enable_log_forwarding or config.enable_monitoring else
```

Comment:
This section is getting convoluted. It is possible/allowed for `aws_lambda_function_event_invoke_config` to have an empty body, e.g. `'aws_lambda_function_event_invoke_config': {}`? If so, the condition in this line is redundant with the checks you added above and can be removed (thus also removing the need for injecting the `aws_lambda_function_event_invoke_config` key using `**`).

Reply:
>It is possible/allowed for … have an empty body

Hm… Apparently it's ok.